### ISSUE TITLE
Document permissions needed

### ISSUE THREAD

**Issue by cblomart** [pnp/sp-dev-modernization#493]
#### Category

[ ] Bug
[x] Enhancement

#### Problem Area

[ ] Page Transformation: Error during the setup/use of the Page Transformation UI solution (did you check our [troubleshooting guide](https://github.com/SharePoint/sp-dev-modernization/blob/dev/Solutions/PageTransformationUI/docs/troubleshootingguide.md)?)
[ ] Page Transformation: Error during the use of page transformation from PnP PowerShell
[ ] Page Transformation: Error during the use of page transformation from .Net
[ ] Page Transformation: Page is not looking correct after transformation
[x] Modernization Scanner: something went wrong...

#### Expected or Desired Behavior

The doc mentions that that advised method of authentication is via an Azure AD app.
The documented permissions are "sites.fullcontrol.all" as app permission and "user.read" as a delegated permission.

We intend to use the application for its scanning capabilities and not to transform anything.
I would then expect the permissions to be much more limited.

Is there a possibility to reduce these permissions?

I opened an issue on the doc page first so here is the reference: https://github.com/SharePoint/sp-dev-docs/issues/6022

**Comment by pkbullock** [comment]
Hi

Unfortunately not, this is more of a limitation with the Azure AD app api permission itself that doesn’t offer a tenant level read option or full read of site collections and its configuration.

Paul

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by cblomart** [comment]
A pitty SharePoint and Azure AD doesn't help define this type of app permission (i get that site.read.all would be limited maybe not able to look at configs).

I get we'll have look if we can work with sites.fullcontrol.all then (which is scary every time I write it 😨)

**Comment by pkbullock** [comment]
I agree, wish there were granular permissions,  there are plenty of customers I work with would appreciate the capability as well.

There is the option for setting up a dev tenant or if you have access to a demo tenant and doing a test run to better understand its operation. The source code is all open source so you can see what the scanner does under the hood.

Additionally, once the scan is complete, simply remove the consent on the Azure AD app to block its use when idle or remove once you have completed the scan. Keep any secrets or certificates in a secure location.

Hope this helps and good luck!

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by jansenbe** [comment]
Thanks @pkbullock for the detailed feedback.

@cblomart : next to what Paul already mentioned you can always run the scan with a user account, this way you can scope the permissions to just the sites you want to scan. If you plan to scan all sites then app-only is the recommended option as granting a user account access to all sites is cumbersome.

**Comment by jansenbe** [comment]
Closing this issue as the changing the auth model is not in scope of this project and all the needed information has been provided.
